src/Data/Vec.purs

```diff
@@ -290,16 +291,18 @@ instance eqVec :: (Nat s, Eq a) => Eq (Vec s a) where
   eq (Vec v1) (Vec v2) = v1 == v2
 
 instance showVec :: (Nat s, Show a) => Show (Vec s a) where
-  show (Vec v) = show v
+  show (Vec v) = "(Vec " <> show v <> ")"
```

Comment:
Doesn't this expose the actually hidden implementation to the outside?
If I do `show (vec2 1 2)` I see `Vec [1, 2]` which suggests that there's an array inside.

how about this one:
```
import Data.Typelevel.Num.Sets (toInt)
import Data.Typelevel.Undefined (undefined)

[...]
instance showVec :: (Nat s, Show a) => Show (Vec s a) where
  show (Vec xs) =
    "(" <>
      String.joingWith " " (["vec" <> show $ toInt (undefined :: s)] <> map show xs)
      <> ")"
```

should give me `show (vec2 1 2)` for `show (vec2 1 2)` (untested)

Reply:
Ah, good point, I'll look into it. So what you mean is that `show (vec2 1 2)` should give us `"(vec2 1 2)"`? The problem i see with this is that it wouldn't always output "real code", because if we had a vector with four elements, it would use `vec4`, even though no such function exists. I'd instead suggest that `show (vec2 1 2)` would give `"(1 +> 2 +> empty)"`, which will produce "real code" for a vector of any length.

Reply:
went ahead and implemented it like i suggested at the end